utils/zerovec/src/ule/slices.rs

```diff
@@ -33,3 +33,25 @@ unsafe impl VarULE for str {
         str::from_utf8_unchecked(bytes)
     }
 }
+
+// Safety (based on the safety checklist on the VarULE trait):
+//  1. [u8] does not include any uninitialized or padding bytes (achieved by being a slice of a ULE type)
+//  2. [u8] is aligned to 1 byte (achieved by being a slice of a ULE type)
+//  3. The impl of `validate_byte_slice()` returns an error if any byte is not valid.
+//  4. The impl of `validate_byte_slice()` returns an error if the slice cannot be used in its entirety
+//  5. The impl of `from_byte_slice_unchecked()` returns a reference to the same data.
+//  6. All other methods are defaulted
+//  7. `[u8]` byte equality is semantic equality
+unsafe impl VarULE for [u8] {
```

Comment:
I re-added this impl, because `[u8]` is a very common zero-copy type that has been established as such before this crate existed. I decided against having an impl for all `[T: ULE]` since it feels like a footgun.

Reply:
Good plan!